include/net/ieee802154_radio.h

```diff
@@ -101,6 +122,16 @@ extern enum net_verdict ieee802154_radio_handle_ack(struct net_if *iface,
  */
 void ieee802154_init(struct net_if *iface);
 
+/**
+ * @brief Check if AR flag is set on the frame inside given net_pkt
+ *
+ * @param pkt A valid pointer on a net_pkt structure, must not be NULL.
+ *
+ * @return True if AR flag is set, False otherwise
+ */
+bool ieee802154_is_ar_flag_set(struct net_pkt *pkt);
```

Comment:
Not clear to me what the compiler does in this case. It seems that if we are incuring a function call overhead for each time we check this bit that it would be better to just make this an inline/macro. Although I do understand that would require exposing the details of the 802.15.4 protocol frame structure at the /include level... but I'm for doing that anyway.